internal/confmapprovider/discovery/discoverer.go

```diff
@@ -117,6 +119,38 @@ func newDiscoverer(logger *zap.Logger) (*discoverer, error) {
 	return d, nil
 }
 
+func (d *discoverer) discoveryAuthConfFromEnv(discoveryReceiverRaw map[string]any) (*confmap.Conf, error) {
+	yamlProvider := yamlprovider.New()
+	envProvider := envprovider.New()
+
+	providers := map[string]confmap.Provider{
+		"yaml": yamlProvider,
+		"env":  envProvider,
+	}
+	out, err := yaml.Marshal(discoveryReceiverRaw)
+	if err != nil {
+		d.logger.Warn(fmt.Sprintf("Failed to create resolver from yaml provider. %v", err))
```

Comment:
this is something we need to bail on:

```suggestion
	if err != nil {
		return nil, fmt.Errorf("failed to marshal discovery receiver config for uri: %w", err)
```

Reply:
:done